core/src/main/scala/com/avast/clients/rabbitmq/RabbitMQConnection.scala

```diff
@@ -105,8 +106,17 @@ object RabbitMQConnection extends StrictLogging {
     Resource.make {
       Sync[F].delay {
         import connectionConfig._
-
-        val factory = new ConnectionFactory
+        import com.avast.clients.rabbitmq.AddressResolverType._
+        val factory = new ConnectionFactory {
+          override def createAddressResolver(addresses: util.List[Address]): AddressResolver = addressResolverType match {
+            case Default => super.createAddressResolver(addresses)
```

Comment:
I'm not sure about this. It basically means "keep it up to the underlying implementation", where I'd expect some specific value instead.

Reply:
Yes, I want to have this `Default` value so the migration will be smooth - we will not break anything.